Embedded domain field into unstructured 2D mesh struct.

sriharshakandala opened the pull request.

jakebolewski commented:
@sriharshakandala I think this needs to be rebased after https://github.com/CliMA/ClimaCore.jl/pull/201 is merged.  Looking at it though I don't think it will be too bad, also #201 allows for the coordinate types defined in #81 to be applied for the sphere.

jakebolewski commented on src/Meshes/Meshes.jl:
```diff
@@ -184,7 +183,9 @@ https://p4est.github.io/papers/BursteddeWilcoxGhattas11.pdf
 $(DocStringExtensions.FIELDS)
 """
-struct Mesh2D{I, IA1D, IA2D, FT, FTA2D, SNT, NB} <: AbstractMesh{FT}
+struct Mesh2D{I, IA1D, IA2D, FT, FTA2D, SNT, NB, D} <: AbstractMesh{FT}
```
usually the unique types are in the order of the fields to make it easy to reference

jakebolewski commented on src/Domains/Domains.jl:
```diff
@@ -125,14 +125,22 @@ end
     Geometry.Cartesian2DPoint{FT}
 
+abstract type AbstractSphere{FT} end
+struct EquiangularSphere{FT} <: AbstractSphere{FT} end
```
isn't the sphere type more a property of the mesh than the domain?

jakebolewski replied:
ex. an `EquiangularSphereMesh()` makes sense but a equiangular sphere domain does not really as the domain just defines the coordinates (LatLong) / coordinate extent (Radius).

sriharshakandala commented:
@jakebolewski : I rebased the code with master. I also added the bickley jet CG experiment with unstructured mesh to the buildkite so that changes relevant to unstructured mesh will also be carried over wherever newer PRs are added.

sriharshakandala commented:
bors try

jakebolewski commented:
Nice, seems to be a bit simpiler and is more in line with what we do for the vertical warping. LGTM

sriharshakandala commented:
Bots r+
